### PR TITLE
CP-35478: Update branding

### PR DESCRIPTION
**Pull request by jijiang**
Signed-off-by: Ji Jiang <ji.jiang@citrix.com>

### CODE REVIEW

**Review by AaronRobson — APPROVED** [review]

File: XenAdmin/Program.cs [hunk 1]
```diff
@@ -429,12 +429,12 @@ private static void ConnectPipe()
 
                     if (firstArgType == ArgType.Passwords)
                     {
-                        log.Error("Refusing to accept passwords request down pipe.  Use XenCenterMain.exe directly");
+                        log.Error("Refusing to accept passwords request down pipe.  Use CitrixHypervisorManagementMain.exe directly");
```
Comment (kc284) [inline XenAdmin/Program.cs:432]: `log.ErrorFormat("Refusing to accept passwords request down pipe. Use {0}Main.exe directly", BrandManager.BRAND_CONSOLE.Replace(" ",""));`
Same below.

File: splash/main.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ const TCHAR SplashClassName[] = TEXT("XenCenterSplash0001");
 const TCHAR PipeStub[] = TEXT("\\\\.\\pipe\\XenCenter-");
 const TCHAR SplashPipeStub[] = TEXT("\\\\.\\pipe\\XenCenterSplash-");
 // The path to the main C# XenCenter exe, relative to the location of the splash exe.
-const TCHAR XenCenterPath[] = TEXT("XenCenterMain.exe");
+const TCHAR XenCenterPath[] = TEXT("CitrixHypervisorManagementMain.exe");
```
Comment (kc284) [inline splash/main.cpp:84]: We can ise a variable here like in line 91.

File: scripts/xenadmin-build.sh [hunk 3]
```diff
@@ -68,31 +68,31 @@ cd ${REPO}
...
 cd ${REPO} && "${MSBUILD}" ${SWITCHES} XenAdmin.sln
 
 #sign files only if all parameters are set and non-empty
 SIGN_BAT="${REPO}/scripts/sign.bat"
-SIGN_DESCR="${BRANDING_COMPANY_NAME_SHORT} ${BRANDING_BRAND_CONSOLE}"
+SIGN_DESCR="${BRANDING_COMPANY_NAME_SHORT} ${BRANDING_BRAND_CONSOLE_NO_SPACE}"
 
 if [ -f "${SIGN_BAT}" ] ; then
-  for file in XenCenterMain.exe CommandLib.dll MSTSCLib.dll XenCenterLib.dll XenModel.dll XenOvf.dll XenOvfTransport.dll
+  for file in CitrixHypervisorManagementMain.exe CommandLib.dll MSTSCLib.dll CoreUtilsLib.dll XenModel.dll XenOvf.dll XenOvfTransport.dll
```
Comment (kc284) [inline scripts/xenadmin-build.sh:82]: We can use `${BRANDING_BRAND_CONSOLE_NO_SPACE}` here and below.

File: scripts/xenadmin-build.sh [hunk 4]
```diff
@@ -152,73 +152,73 @@ locale_id() {
...
 echo "INFO: Build the tests..."
 cd ${REPO}/XenAdminTests && "${MSBUILD}" ${SWITCHES}
 cp ${REPO}/XenAdmin/ReportViewer/* ${REPO}/XenAdminTests/bin/Release/
 cd ${REPO}/XenAdminTests/bin/ && tar -czf ${OUTPUT_DIR}/XenAdminTests.tgz ./Release
-cd ${REPO}/XenAdmin/TestResources && tar -cf ${OUTPUT_DIR}/${BRANDING_BRAND_CONSOLE}TestResources.tar *
+cd ${REPO}/XenAdmin/TestResources && tar -cf ${OUTPUT_DIR}/${BRANDING_BRAND_CONSOLE_NO_SPACE}TestResources.tar *
 
 #include cfu validator binary in output directory
-cd ${REPO}/CFUValidator/bin/Release && zip CFUValidator.zip ./{*.dll,CFUValidator.exe,XenCenterMain.exe}
+cd ${REPO}/CFUValidator/bin/Release && zip CFUValidator.zip ./{*.dll,CFUValidator.exe,CitrixHypervisorManagementMain.exe}
```
Comment (kc284) [inline scripts/xenadmin-build.sh:210]: `${BRANDING_BRAND_CONSOLE_NO_SPACE}` and below.

File: XenAdmin/XenAdmin.csproj [hunk 5]
```diff
@@ -9,7 +9,7 @@
     <OutputType>WinExe</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>XenAdmin</RootNamespace>
-    <AssemblyName>XenCenterMain</AssemblyName>
+    <AssemblyName>CitrixHypervisorManagementMain</AssemblyName>
```
Comment (kc284) [inline XenAdmin/XenAdmin.csproj:12]: `[XenCenter]Main` (I think it should compile); then rebrand this file with rebranding_global in the build scripts.

**Review by kc284 — CHANGES_REQUESTED** [review]

File: WixInstaller/branding.wxs [hunk 6]
```diff
@@ -34,44 +34,6 @@
 
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:netfx="http://schemas.microsoft.com/wix/NetFxExtension">
     <?include branding.wxi ?>
```
Comment (kc284) [inline WixInstaller/branding.wxs:36]: This file does not need to change since it is overwritten by the internal.

File: WixInstaller/branding.wxi [hunk 7]
```diff
@@ -35,25 +35,25 @@
...
-  <?define TestResourcesGuid="FA8D4F56-A94A-467c-9E6B-F3DC26F95B1E"?>
-  <?define ExtToolsGuid="D5FC0252-C97B-46e7-9633-A6B68EDB6654"?>
-  <?define SchemasFilesGuid="E2186CD8-5064-4414-8AD7-E4495B6A3204"?>
-  <?define RegEntriesGuid="193BAE1F-F2AE-4451-94DC-4B105DB5179C"?>
-  <?define AppShortcutGuid="6B875059-26BC-4fa7-ACB7-0B9A4E4665CA"?>
-  <?define ReadmeGuid="47427a60-4064-4fdb-878d-04309a0fd9ce"?>
-  <?define UpdateFilesGuid="1cfbf607-cc80-4bf8-b2fc-37e69c872316"?>
-  <?define HealthCheckGuid="{9D686BFC-B4FD-435F-AC74-0ACE29425095}"?>
-  <?define DummyGuid="98945751-78CD-421F-9478-352AF41F496C"?>
+  <?define UpgradeCode="{5b42ef0d-e5b3-4538-bd2b-a94a77023e39}"?>
```
Comment (kc284) [inline WixInstaller/branding.wxi:38]: These IDs should not change. The file is overwritten by the internal.

File: WixInstaller/branding.wxs (hunk 6 above)
Comment (kc284) [reply WixInstaller/branding.wxs:36]: I forgot: we do need checks in the code to only use the healthcheck for Stockholm and earlier hosts.

File: WixInstaller/XenCenter.wxs [hunk 8]
```diff
@@ -35,13 +35,13 @@
...
-    <Product Id="$(var.ProductCode)" Name="$(var.CompanyNameShort) $(var.BrandConsole)" Language="$(env.WixLangId)" Version="$(var.ProductVersion)" Manufacturer="$(var.CompanyNameLegal)" UpgradeCode="$(var.UpgradeCode)">
-        <Package Languages="1033,1041,2052,1028" Description="$(var.CompanyNameShort) $(var.BrandConsole)" Comments="none." InstallerVersion="200" Compressed="yes" />
+    <Product Id="$(var.ProductCode)" Name="$(var.BrandConsole)" Language="$(env.WixLangId)" Version="$(var.ProductVersion)" Manufacturer="$(var.CompanyNameLegal)" UpgradeCode="$(var.UpgradeCode)">
+        <Package Languages="1033,1041,2052,1028" Description="$(var.BrandConsole)" Comments="none." InstallerVersion="200" Compressed="yes" />
         <Media Id="1" Cabinet="XenCenter.cab" EmbedCab="yes" />
         <Directory Id="TARGETDIR" Name="SourceDir">
             <Directory Id="ProgramFilesFolder">
                 <Directory Id="Citrix" Name="$(var.CompanyNameShort)">
-                    <Directory Id="INSTALLDIR" ShortName="XenCente" Name="$(var.BrandConsole)">
+                    <Directory Id="INSTALLDIR" ShortName="CHM" Name="$(var.BrandConsoleNoSpace)">
```
[the remark this answers is not in the document]
Comment (jijiang) [reply WixInstaller/XenCenter.wxs:44]: I've added new variable var.BrandConsoleShort. I'll raise a PR to add it to branding.git repo.

File: XenAdmin/XenAdmin.csproj (hunk 5 above)
Comment (jijiang) [reply XenAdmin/XenAdmin.csproj:12]: I've tried to use [XenCenter_No_Space]Main here, but this is not replaced with "CitrixHypervisorManagementMain" during the build. ([XenCenter_No_Space] in other files could be replaced). So I think we have to hardcode here.

Comment (kc284) [reply XenAdmin/XenAdmin.csproj:12]: Check whether Xenadmin.csproj is run through rebranding_global in the rebranding.sh.

Comment (jijiang) [reply XenAdmin/XenAdmin.csproj:12]: ok, it works now!

File: WixInstaller/branding.wxs (hunk 6 above)
Comment (jijiang) [reply WixInstaller/branding.wxs]: In my understanding, Stockholm and earlier builds won't trigger the copy of the file from xenadmin-branding repo, therefore I just don't need to touch this file in xenadmin repo. Is it right?

**Review by kc284 — APPROVED** [review]

**Review by kc284 — APPROVED** [review]

**Review by MihaelaStoica — APPROVED** [review]

**Pull request merged**